src/v/cluster/controller_log_limiter.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/commands.h"
+#include "config/configuration.h"
+#include "ssx/metrics.h"
+#include "utils/token_bucket.h"
+#include "vlog.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <optional>
+
+namespace cluster {
+
+inline ss::logger controller_rate_limiter_log("controller_rate_limiter_log");
+
+struct limiter_configuration {
+    config::binding<bool> enable;
+    config::binding<size_t> topic_rps;
+    config::binding<std::optional<size_t>> topic_capacity;
+    config::binding<size_t> acls_and_users_rps;
+    config::binding<std::optional<size_t>> acls_and_users_capacity;
+    config::binding<size_t> node_management_rps;
+    config::binding<std::optional<size_t>> node_management_capacity;
+    config::binding<size_t> move_rps;
+    config::binding<std::optional<size_t>> move_capacity;
+    config::binding<size_t> configuration_rps;
+    config::binding<std::optional<size_t>> configuration_capacity;
+};
+
+class group_limiter {
+public:
+    explicit group_limiter(
+      ss::sstring group_name,
+      config::binding<size_t> rate_binding,
+      config::binding<std::optional<size_t>> capacity_binding);
+
+    bool try_throttle();
+
+private:
+    void update_rate();
+    void update_capacity();
+
+    ss::sstring _group_name;
+    config::binding<size_t> _rate_binding;
+    config::binding<std::optional<size_t>> _capacity_binding;
+    token_bucket<> _throttler;
+};
+
+class controller_log_limiter {
+public:
+    explicit controller_log_limiter(limiter_configuration configuration);
+
+    template<typename Cmd>
+    bool throttle(const Cmd& cmd) {
+        if (!_enabled()) {
+            return true;
+        }
+        switch (cmd.batch_type) {
+        case model::record_batch_type::topic_management_cmd:
+            switch (cmd.type) {
+            case create_topic_cmd_type:
+            case delete_topic_cmd_type:
+            case update_topic_properties_cmd_type:
+            case create_partition_cmd_type:
+            case create_non_replicable_topic_cmd_type:
+                return _topic_operations_limiter.try_throttle();
+            case move_partition_replicas_cmd_type:
+            case cancel_moving_partition_replicas_cmd_type:
```

Comment:
Initially I thought that we also want to limit these requests
As we have separate quota for move_partitions and cancel movement, for users it would be hard to consume all limit.
This quota will better work for movements from internal requests like partition balancer perform.